### ISSUE TITLE
Make mentoring matter

### ISSUE THREAD

**Issue by asarkar** [exercism/exercism#6387]
To prevent the platform from becoming another Stackoverflow, where people dump their questions showing absolutely zero effort, and walk away to go have a beer, I recommend that every successful mentoring session should deduct either some points from the student's reputation, or require actual $$ (for those not having enough rep or not willing to shed some). Either of those would go directly to the mentor. The only way to not pay would be to dispute a mentoring session, which is when admin can get involved. The obvious benefits of this process are:
1. Weed out zero-effort mentoring requests.
2. Free up mentors' time for those who really need help.
3. Encourage mentors to participate more because it would directly benefit them.

**Comment by asarkar** [comment]
@jonathandmiddleton I don't have visibility into mentoring results at Exercism, but as I stated above, it is all too common at Stackoverflow. I should know, I've been on SO long enough to know and have almost 20k reputation. Much like natural resources, if something is free and available, it is human tendency to abuse it.
I'm going to turn the table and ask you the same question: How do you determine the effort put in by the user before they request a mentor?

**Comment by glennj** [comment]
My experience as a longtime mentor (and 200k SO rep user). Stay on Stack Overflow long enough and it becomes pretty easy to sniff out the homework questions. Here on exercism, it truly seems to be people focused on learning the language and seeking expert advice.

* With the web editor and test runners of Exercism v3, students who choose not to work locally have to get their code passing the test suite before they request mentoring. So that demonstrates significant effort.
* Community solutions can be found, even if they are supposed to be "locked". If a person decides to cheat and copy someone else's code, that only harms the student.
* @iHiD posted some stats on slack recently. With v3 versus v2, there are roughly 4x more solutions submitted per month. Additionally, to quote:
    * There have been 1/3 as many discussions in v3 as the previous 8 months in v2
    * There are 43.10% as many profitable discussions in v3 as v2 (where "profitable" means the student actually engages after the mentor posts)
    * There are 6.5x less "wasted" discussions (where the student doesn't reply to the mentor) - which equates to 60 fewer wasted discussions per day. Only 16% of discussions do not elicit student engagement now.

 The mission statement is a work in progress, but one of the core tenets is "supporting free lifelong education" so IMO charging $ to jump the queue will be a non-starter.

**Comment by iHiD** [comment]
While we're always focussed on improving the experience of both students and mentors, as there's no reason to think this is a prolific issue at Exercism (based on either mentors' experience or data), and as we've already made significant steps forward with this area in v3, I'm going to close this. 

We're definitely not going to charge students, and reputation is an incrementing system. We did consider other systems (such as accumulating and using credits) but removed it before launching v3 as we had no data to consider that it was solving a problem. 

Fundamentally, I like us building a system where anyone can ask for help respectfully, and anyone else can give their help generously, without creating some sort of barter or payment exchange. And so until this is actually a problem that exists its not something I see us putting energy into trying to solve.
